pkg/etcd/etcd.go

```diff
@@ -1074,7 +1074,8 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 
 		nodes, err := e.getETCDNodes()
 		if err != nil {
-			logrus.Warnf("Failed to list nodes with etcd role: %v", err)
+			logrus.Errorf("Failed to list nodes with etcd role: %v", err)
+			continue
```

Comment:
It looks like doing this has broken member promotion - CI shows that members are not getting promoted up to voting member.

We need to skip the node condition handling in a way that doesn't prevent member promotion from occurring.

Reply:
I think we could just remove this one and be okay with it, since this will work and prevent from setting nil in the node variable

```golang
if node == nil {
    continue
}
```

Reply:
right, but we can't continue there either, because that will skip the leader progress tracking - which does not need a node reference at all. See my most recent comments.